pkg/sql/sqlbase/rowwriter.go

```diff
@@ -99,7 +99,7 @@ func (rh *rowHelper) encodeIndexes(
 func (rh *rowHelper) encodeSecondaryIndexes(
 	colIDtoRowIndex map[ColumnID]int, values []tree.Datum,
 ) (secondaryIndexEntries []IndexEntry, err error) {
-	if len(rh.indexEntries) != len(rh.Indexes) {
+	if rh.indexEntries == nil {
```

Comment:
Given that we do have expected cases of rh.indexEntries != nil I feel more comfortable with you leaving the code as is ```if len(rh.indexEntries) != len(rh.Indexes) ```so as to never trigger the panic you've added to the code.